internal/venice-test-common/src/main/java/com/linkedin/venice/utils/TestUtils.java

```diff
@@ -778,18 +778,45 @@ public static void shutdownThread(Thread thread, long timeout, TimeUnit unit) th
     Assert.assertFalse(thread.isAlive());
   }
 
+  /**
+   * calls {@link ExecutorService#shutdownNow} which stops accepting new tasks and interrupts the running tasks
+   */
+  public static void shutdownExecutorNow(ExecutorService executor) throws InterruptedException {
```

Comment:
@anaberezhnov This was added as a part of this [commit](https://github.com/linkedin/venice/commit/c276e2fbb623a890d658f47bb00b827ffcf99f76) to `Cleanup Thread/ThreadPool leaks`. Please help review whether these changes might break anything related to that.

Reply:
According to `RocksDBThrottlerTest::testThrottle` it's not waiting and shutting down thread pool immediately. Also `testThrottle` looks deficient: I don't see any assert in it. What is it even testing?

Reply:
As discussed offline, `RocksDBThrottlerTest::testThrottle` seems to be incomplete. cleaned it up further into 2 different tests as shown below. 
1. `testThrottleLiveness`: Testing whether all the tasks invoking the throttlers gets executed
2. `testThrottleCorrectness`: Testing whether only the allowedMaxOperationsInParallel are executed at any time.

Reply:
reset these changes and added a comment for `shutdownExecutor()` that it calls `ExecutorService#shutdownNow` to avoid being mistaken.